src/init.cpp

```diff
@@ -993,9 +994,10 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     if (!mapMultiArgs["-bip9params"].empty()) {
         // Allow overriding BIP9 parameters for testing
-        if (!Params().MineBlocksOnDemand()) {
-            return InitError("BIP9 parameters may only be overridden on regtest.");
+        if (!Params().AllowsOverriddenSoftFork()) {
```

Comment:
He is creating a new one. Maybe a future testnet allows this but does not mine blocks on demand. That's why we have different properties for chainparams instead of calling IsTestnet() or IsRegtest () like we did before.